staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -160,7 +160,10 @@ type watchCache struct {
 	// you should always apply modulo capacity to get an index in cache array.
 	cache      []*watchCacheEvent
 	startIndex int
-	endIndex   int
+	watchSize  int
```

Comment:
Let me understand your problem better - are you afraid of startIndex/endIndex overflow in int64?

If so - I'm reluctant to accept this change - as an example etcd RV is also int64 and can potentially also overflow and I'm sure it would cause much more pain for us than that.
But realistically - even if we assume 1000 items/s churn, that means we have 10^15 seconds until it will overflow, which is millions of years...